From 6f39c7147fc306de34dd3fa351b398de68de3ea6 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sat, 31 Mar 2018 10:06:48 +0200
Subject: [PATCH 2/4] wacom: Report touch when pen button is pressed if
 gestures are off

* Some applications, like Squid, allow interpreting input differently
  when the pen button is pressed, so allow users to keep that behavior

Change-Id: Ifce2f509646c7676c931f4f04ff482acdf5ac990
Signed-off-by: Paul Keith <javelinanddart@gmail.com>
---
 drivers/input/touchscreen/wacom/wacom_i2c_func.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/wacom/wacom_i2c_func.c b/drivers/input/touchscreen/wacom/wacom_i2c_func.c
index f9250eeba00..7c868607aae 100644
--- a/drivers/input/touchscreen/wacom/wacom_i2c_func.c
+++ b/drivers/input/touchscreen/wacom/wacom_i2c_func.c
@@ -1113,7 +1113,7 @@ int wacom_i2c_coord(struct wacom_i2c *wac_i2c)
 #endif
 			input_report_key(wac_i2c->input_dev,
 					 BTN_STYLUS, stylus);
-			if (!stylus && !wac_i2c->side_pressed) {
+			if (!wac_i2c->enabled_gestures || (!stylus && !wac_i2c->side_pressed)) {
 				input_report_key(wac_i2c->input_dev, BTN_TOUCH, prox);
 			}
 			input_report_key(wac_i2c->input_dev, wac_i2c->tool, 1);
-- 
2.15.1

